packages/core/src/integrations/webpack/factory.ts

```diff
@@ -0,0 +1,86 @@
+import type {
+  AsyncContainer,
+  IRemoteScriptFactory,
+  ISharingScopeFactory,
+  RemoteOptions,
+  SharedScope,
+} from '../../types';
+import type { WebpackRequire } from './types';
+
+export class WebpackSharingScopeFactory implements ISharingScopeFactory {
+  async initializeSharingScope(scopeName = 'default'): Promise<SharedScope> {
+    const webpackShareScopes =
+      __webpack_share_scopes__ as unknown as SharedScope;
+
+    if (!webpackShareScopes?.default) {
+      await __webpack_init_sharing__(scopeName);
+    }
+
+    // TODO: Why would we reference __webpack_require and not __webpack_share_scopes__ ?
+    return (__webpack_require__ as unknown as WebpackRequire)
+      .S as unknown as SharedScope;
+  }
+}
+
+export class WebpackRemoteScriptFactory implements IRemoteScriptFactory {
+  loadScript(containerKey: string, remoteOptions: RemoteOptions) {
+    return webpackLoadScript(containerKey, remoteOptions.url);
+  }
+}
+
+const webpackLoadScript = (
+  containerKey: string,
+  url: string
+): AsyncContainer => {
+  const scope =
```

Comment:
I think we should use `getScope` here.

Reply:
Updated. The intent with the "integrations" directory is that these may be build plugins at some point. Still don't have a design I am sold on, but eventually these could be great for how you would extend the runtime.